lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -765,7 +834,41 @@ namespace Js
         CleanupBeforeExecution();
 
         Arguments outArgs(CallInfo(CallFlags_Value, 0), nullptr);
-        return rootFunction->CallRootFunction(outArgs, scriptContext, true);
+        
+        Var ret = nullptr;
+        try
+        {
+            AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_JavascriptException));
+            ENTER_SCRIPT_IF(scriptContext, true, false, false, !scriptContext->GetThreadContext()->IsScriptActive(),
+            {
+                ret = rootFunction->CallRootFunction(outArgs, scriptContext, true);
+            });
+        }
+        catch (const Js::JavascriptException& err)
+        {
+            Var errorObject = err.GetAndClear()->GetThrownObject(scriptContext);
+            AssertMsg(errorObject != nullptr && Js::JavascriptError::Is(errorObject), "ModuleEvaluation: Unhandled exception thrown by calling root function");
+            if (errorObject != nullptr && Js::JavascriptError::Is(errorObject))
+            {
+                ResolveOrRejectDynamicImportPromise(false, errorObject, scriptContext);
```

Comment:
Adding 'this' as last param and SetPromise(nullptr) will be taken care of.

---
In reply to: [115383242](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115383242) [](ancestors = 115383242)